scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -473,24 +478,37 @@ def load_quarantine(self, file):
         # Load yaml into quarantine_yaml
         quarantine_yaml = scl.yaml_load_verify(file, self.quarantine_schema)
 
+
+        self.quarantine.platforms = quarantine_yaml.get('platforms', [])
+        self.quarantine.architectures = quarantine_yaml.get('architectures', [])
+        self.quarantine.simulation = quarantine_yaml.get('simulation', [])
+
         # Create quarantine_list with a product of the listed
         # platforms and scenarios for each entry in quarantine yaml
         quarantine_list = []
-        for quar_dict in quarantine_yaml:
+        for quar_dict in quarantine_yaml['tests']:
             if quar_dict['platforms'][0] == "all":
                 plat = self.platform_names
             else:
                 plat = quar_dict['platforms']
                 self.verify_platforms_existence(plat, "quarantine-list")
             comment = quar_dict.get('comment', "NA")
-            quarantine_list.append([{".".join([p, s]): comment}
-                                   for p in plat for s in quar_dict['scenarios']])
+            scenarios = []
+            for _, ts in self.testsuites.items():
+                scenarios.append(ts.id)
+            q_scenarios = []
+            for s in quar_dict['scenarios']:
+                r = re.compile(s)
+                q_scenarios.extend(list(filter(r.fullmatch, scenarios)))
+
+            quarantine_list.append([{"/".join([p, s]): comment}
+                                   for p in plat for s in q_scenarios])
 
         # Flatten the quarantine_list
         quarantine_list = [it for sublist in quarantine_list for it in sublist]
         # Change quarantine_list into a dictionary
         for d in quarantine_list:
-            self.quarantine.update(d)
+            self.quarantine.tests.update(d)
```

Comment:
We store here a list of all possible configurations to be quarantined (a Cartesian product of scenarios and platform), the list can be very large. Maybe it could be enough to keep in Quarantine class only data from yaml file and do filtering in `apply_filters` method (like for newly added platforms, architectures ...).